sympy/functions/special/delta_functions.py

```diff
@@ -99,16 +101,17 @@ def simplify(self, x):
         """
         from sympy.polys.polyroots import roots
 
-        if not self.args[0].has(x) or (len(self.args) > 1 and self.args[1] != 0 ):
+        wrt = hints.pop('wrt')
```

Comment:
That looks wrong. Checking if something is in a dictionary shouldn't raise KeyError. Try deleting your pyc files.